### ISSUE TITLE
test using pretrained model fails

### ISSUE THREAD

**Issue by guifentian** [Eaphan/BiProDet#14]
Thank you for shareing.
There are some problems when trying train or test using tools/cfgs/kitti_models/biprodet.yaml. 

A. training: 
1. segmentation data is needed. Even if 'segmentation' removed from the above config, data augmentation using segmentation is hard-coded in the code in datasets_sampler.py

B. Test:
Seems the model shared in google drive only contains weights in lidar parts,  no image backbone or roi parts.

**Comment by guifentian** [comment]
BTW, the paper mentioned about 
"we used the instance segmentation annotations as provided in (Qi et al.,
2019)."

where is the code to read from train.json which has intance masks?
when running  the following command, only standard annotation data of 3d box is cared is read in and generate info.pkl files. 
https://github.com/Eaphan/BiProDet/tree/main#prepare-kitti-dataset

but the dataloader of train.py is trying to get "segmentation" dicts which is not in info.pkl

**Comment by NiranjanRavi1993** [comment]
@guifentian  Hi, I am facing the segmentation label issue. I looked into this issue: https://github.com/Eaphan/BiProDet/issues/6 and used the downloaded .pkl files and changed my path as below:
INFO_PATH: {
    'train': [kitti_infos_train.pkl, kitti_infos_val.pkl],
    'test':  [kitti_infos_val.pkl],
}
Still the same error persists. My aim to train the network (with my changes) using train+val to evaluate on KITTI server. What changes should I carry out in current repo to address the above problem?

Thanks in advance. 

Update: readMe had the answers

**Comment by Eaphan** [comment]
> @guifentian Hi, I am facing the segmentation label issue. I looked into this issue: #6 and used the downloaded .pkl files and changed my path as below: INFO_PATH: { 'train': [kitti_infos_train.pkl, kitti_infos_val.pkl], 'test': [kitti_infos_val.pkl], } Still the same error persists. My aim to train the network (with my changes) using train+val to evaluate on KITTI server. What changes should I carry out in current repo to address the above problem?
> 
> Thanks in advance.
> 
> Update: readMe had the answers

Yes, we have updated the model weights as stated in README.md.

**Comment by Eaphan** [comment]
> BTW, the paper mentioned about "we used the instance segmentation annotations as provided in (Qi et al., 2019)."
> 
> where is the code to read from train.json which has intance masks? when running the following command, only standard annotation data of 3d box is cared is read in and generate info.pkl files. https://github.com/Eaphan/BiProDet/tree/main#prepare-kitti-dataset
> 
> but the dataloader of train.py is trying to get "segmentation" dicts which is not in info.pkl

Yes. I wrote a script to match the bounding box with instance segmentation labels in KINS dataset. I will upload once I can get my hands on a hard disk that is not mounted for the moment. But I think you can use the info*pkl files without the scripts.

Can you report information or log about the training error in detail?

**Comment by Eaphan** [comment]
We have provided the script to match bounding box with instance segmentation labels at https://github.com/Eaphan/UPIDet/blob/main/tools/convert_kins_label.py
